dlc-manager/tests/ln_dlc_channel_execution_tests.rs

```diff
@@ -935,7 +939,21 @@ fn ln_dlc_test(test_path: TestPath) {
                 bob_descriptor.clone(),
             );
             off_chain_close_offer(&test_params, &alice_node, &bob_node, channel_id);
-            off_chain_close_finalize(&test_params, &alice_node, &bob_node, channel_id);
+            off_chain_close_finalize(&alice_node, &bob_node, channel_id);
+        }
+
+        if let TestPath::OffChainCloseOpenClose = test_path {
```

Comment:
Wasn't it enough to add an open/close? Are two required? Also I don't think you need another `if let` here?

Reply:
> Wasn't it enough to add an open/close? Are two required?

I could only reproduce the bug by opening, closing, opening and closing a second time. I think it's because the error path can only be hit if there was at least one other secret generated before:

```rust
/// Inserts the `secret` at `idx`. Returns `Ok(())` if the secret
/// was generated in accordance with BOLT 3 and is consistent with previous secrets.
pub fn provide_secret(&mut self, idx: u64, secret: [u8; 32]) -> Result<(), ()> {
    let pos = Self::place_secret(idx);
    // If we place the secret in position 0, the for loop is skipped so there is no error path
    for i in 0..pos {
        let (old_secret, old_idx) = self.old_secrets[i as usize];
        if Self::derive_secret(secret, pos, old_idx) != old_secret {
            return Err(());
        }
    }
    if self.get_min_seen_secret() <= idx {
        return Ok(());
    }
    self.old_secrets[pos as usize] = (secret, idx);
    Ok(())
}
```

> Also I don't think you need another if let here?

Do you mean that this would be the third open-close? I agree, I think there's one cycle too many.

Reply:
> Do you mean that this would be the third open-close? I agree, I think there's one cycle too many.

Yes! And even if there was a need for an extra cycle, the `if` is redundant AFAICT (it's already within an if block with the same condition).

Reply:
That is true (I didn't write this code 😛).